src/Mini-Project-UI.py

```diff
@@ -0,0 +1,198 @@
+    def please_select_directory(self):
+        """Popup to allow the user to select a directory"""
+        pop_up = QtWidgets.QMessageBox()
+        pop_up.setIcon(QtWidgets.QMessageBox.Question)
+        pop_up.setText("Please Select A DICOM File to Open")
+        pop_up.setStandardButtons(QtWidgets.QMessageBox.Close | QtWidgets.QMessageBox.Open)
+        anw = pop_up.exec()
+
+        if anw == QtWidgets.QMessageBox.Close:
+            QtWidgets.QApplication.exit()
+        elif anw == QtWidgets.QMessageBox.Open:
+            dir_path, _ = QtWidgets.QFileDialog.getOpenFileName(self, "Select DICOM File", filter="DICOM Files (*.dcm)")
+            self.path = dir_path
+            self.open_dicom_file()
+
+
+    def file_cannot_be_opened_error(self):
+        error_message = QtWidgets.QMessageBox()
+        error_message.setIcon(QtWidgets.QMessageBox.Warning)
+        error_message.setText("Could not open file sorry, file is either corrupted or incomplete")
+        error_message.exec()
+
+    def directory_button_box(self):
+        """Creates the grid layout for the button box"""
+        self._grid_group_box = QtWidgets.QGroupBox("Patient Information")
+        layout = QtWidgets.QGridLayout()
+        button = QtWidgets.QPushButton("Open New File")
+        path_label = QtWidgets.QLabel("Path: ")
+        self.text = QtWidgets.QLabel()
+        fname_label = QtWidgets.QLabel("First Name:")
+        lname_label = QtWidgets.QLabel("Last Name:")
+        self.fname = QtWidgets.QLabel("First Name")
+        self.lname = QtWidgets.QLabel("Last Name")
+        self.dob_label = QtWidgets.QLabel("Date Of Birth:")
+        self.dob = QtWidgets.QLabel("Date Of Birth: ")
+        self.patient_id_label = QtWidgets.QLabel("patient ID:")
+        self.patient_id = QtWidgets.QLabel("Patient ID")
+        self.sex_label = QtWidgets.QLabel("Sex:")
+        self.sex =  QtWidgets.QLabel("M/F")
+        self.modality_label = QtWidgets.QLabel("Modality:")
+        self.modality = QtWidgets.QLabel("Modality")
+
+        button.clicked.connect(self.add_dic_button)
+        layout.addWidget(path_label, 0, 0)
+        layout.addWidget(self.text, 0, 1, 1,4)
+        layout.addWidget(fname_label, 1, 0)
+        layout.addWidget(lname_label, 2, 0)
+        layout.addWidget(self.dob_label, 3, 0)
+        layout.addWidget(self.sex_label, 1, 3)
+        layout.addWidget(self.patient_id_label, 2, 3)
+        layout.addWidget(self.modality_label, 3, 3)
+        
+        layout.addWidget(self.fname, 1, 1)
+        layout.addWidget(self.lname, 2, 1)
+        layout.addWidget(self.dob, 3, 1)
+        layout.addWidget(self.sex, 1, 4)
+        layout.addWidget(self.patient_id, 2, 4)
+        layout.addWidget(self.modality, 3, 4)
+        layout.addWidget(button, 5, 4)
+        
+        layout.setColumnStretch(1, 20)
+        self._grid_group_box.setLayout(layout)
+
+    def add_dic_button(self):
```

Comment:
what does dic stand for?
you can use longer names for variables and function names.
You write code once, you rewrite it a few times, but you read it *many* times.
Code that can be read and understood from the words it uses is a good thing.
DICOM is not a word, it's an acronym.
dcm is the very commonly used suffix for file names for DICOM objects, and is used frequently as an abbreviation of the DICOM acronym.
You can use "dicom" in lower case to represent DICOM if using snake_case as is typically preferred for Python.